compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1604,25 +1608,34 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
             })
         }
 
+        let real_rust_source_base_dir = sess.real_rust_source_base_dir();
```

Comment:
This fn called *a lot*, you should wrap real_rust_source_base_dir into some once_cell.